reporter-xray-udp/README.md

```diff
@@ -1,3 +1,69 @@
-# reporter-xray-udp
+# AWS X-Ray Reporter for Zipkin
 
-This sends json to XRay's daemon via UDP
\ No newline at end of file
+Used in an application that is being traced, this component converts Zipkin 
+trace data (spans) into [AWS X-Ray](https://aws.amazon.com/xray/)'s
+proprietary format and sends them to the 
+[X-Ray daemon](https://docs.aws.amazon.com/xray/latest/devguide/xray-daemon.html) 
+using UDP.
+
+Note that, unlike `AsyncReporter`, this reporter attempts to encode and send
+the span immediately on the calling thread.
+As UDP is used, there is no latency in waiting for the daemon to accept and 
+respond to the data.
+ 
+## Configuration
+
+To use the X-Ray Reporter, you'll need to:
+
+### 1. Include the dependency in your application
+
+The coordinates of the dependency to include are:
+
+`io.zipkin.aws:zipkin-reporter-xray-udp:<VERSION>`
+
+Note that this also brings in the required `zipkin-storage-xray-udp` dependency.
+
+### 2. Create a Tracing instance that uses the Reporter
+
+For example, if you're tracing in a Spring application, you can configure the
+Tracing bean as such:
+
+```$java
+    @Bean
+    fun tracing(@Value("\${spring.application.name:spring-tracing}") serviceName: String, 
+                spanReporter: Reporter<Span>): Tracing {
+        return Tracing.newBuilder()
+            .localServiceName(serviceName)
+            .spanReporter(XRayUDPReporter.create())
+            .traceId128Bit(true) // X-Ray requires 128-bit trace IDs
+            .sampler(Sampler.ALWAYS_SAMPLE) // Configure as desired
+            .build()
+    }
+```
+
+### 3. Ensure the X-Ray daemon is running alongside your application
+
+The Reporter doesn't send tracing data directly to the X-Ray product, 
+but to an X-Ray daemon accessible locally via UDP.
+You'll need to ensure your application has access to a running X-Ray daemon
+for tracing data to reach X-Ray. AWS provides documentation for
+[coniguring the X-Ray daemon in various environments](https://docs.aws.amazon.com/xray/latest/devguide/xray-daemon.html). 
+Don't forget to give the infrastructure hosting your X-Ray daemon
+[permission to upload traces to X-Ray in IAM](https://docs.aws.amazon.com/xray/latest/devguide/xray-permissions.html).
+
+## Configuring the X-Ray Daemon address
+
+The default address used to send to the daemon is `localhost:2000`.
+
+If you need to override the default address, either:
+1. set the environment variable `AWS_XRAY_DAEMON_ADDRESS`; or
+2. pass the address as a string to `XRayUDPReporter.create()`.
+
+
+
+## AWS Propagation
+
+If you are using components of AWS that natively send tracing to X-Ray
+(as opposed to only tracing with Zipkin throughout the stack), 
+you will want to consider switching to 
+[AWS Trace Propagation](https://github.com/openzipkin/zipkin-aws/tree/master/brave-propagation-aws). 
```

Comment:
I think you actually need to. The impression I've gotten is that XRay will drop traces that are too old, the XRay header encodes the start time in the upper ~32bits. So using the standard zipkin headers will result in random traces being dropped

Reply:
My reading of https://github.com/openzipkin/brave/pull/509 and the code in https://github.com/openzipkin/brave/blob/master/brave/src/main/java/brave/internal/Platform.java are that Zipkin's standard 128-bit TraceIDs are X-Ray friendly. So, a stack that purely uses Zipkin but wants to log into X-Ray only needs the 128-bit IDs and can happily propagate using B3 headers. But if there's AWS components tracing in the stack (e.g. Lambda), then you need the AWS-specific trace propagation.

Perhaps it's worth adding a little more detail about the choice / trade-off?

Reply:
Could you document it a bit clearer whether `zipkin-reporter-xray-udp` can be used in a client side without provisioning a zipkin server?

Reply:
> Could you document it a bit clearer whether `zipkin-reporter-xray-udp` can be used in a client side without provisioning a zipkin server?

Added this: "No Zipkin server is required to use this component with X-Ray. The purpose of the library is to allow applications that trace using Brave's Tracer API and B3 propagation to forward recorded telemetry to X-Ray from the traced applications."

Reply:
Again sorry for the rush. I forgot that was the native 128 bit impl. Would be good to note that 128 bit trace Ids are required in order to report to XRay